### ISSUE TITLE
showInputBox - initial value is not returned, if user presses enter without making a change

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#11503]
```
vscode.window.showInputBox({ value: 'test' }).then(text => {
    console.log('result: ' + text);
});
```

If the user presses enter without typing, '' is returned. If they type anything, the correct value is returned.

OSX, 1.5.0-insider but not 1.4.

**Comment by roblourens** [comment]
7ba701b?

**Comment by riteshatsencha** [comment]
Running into same issue. Version 1.5 on OS X 10.10.3. Default value is not returned if user does not type in the input box. Thank you.

**Comment by golf1052** [comment]
Also noticed this as well when working on my extension.
Version 1.5.2
OS: Windows 10 v1607

**Comment by riteshatsencha** [comment]
Is this issue fixed? Does "closed" mean it's fixed? I still am experiencing the same issue, where I have to modify the default value before the input is accepted. Any thoughts?

**Comment by bpasero** [comment]
@riteshatsencha fixed for 1.6.0 which will be released end of week.

**Comment by vrachels** [comment]
Not sure if this needs a new bug report, or reopen this one.   When pressing enter with no input, if there is a validateInput function, if the default value is invalid, then the default value is still passed as an accepted return value.
ie:
```
const descStr = await vscode.window.showInputBox({
  validateInput: (value: string) => {
    if(!value || value.trim().length === 0) {
      return 'Cannot set empty description';
    }
    return null;
  }
});
if (descStr === undefined || descStr.trim().length == 0) {
  // pressing enter with no other input will still submit the empty string
  Display.showError('Cannot set empty description');
  return;
}
```
expected behavior would be to run the validator first and respond as per usual valid state (prevent acceptance)  (alternately, react as if cancelled)

editing the input box then deleting it again to become invalid works as expected
